decoder/vaapiDecoderJPEG_unittest.cpp

```diff
@@ -139,24 +403,41 @@ VAAPIDECODER_JPEG_TEST(Factory)
     doFactoryTest(mimeTypes);
 }
 
-VAAPIDECODER_JPEG_TEST(Decode_Simple)
+template <std::size_t N>
+void decodeJpeg(const std::tr1::array<uint8_t, N>& data, uint32_t fourcc)
 {
     VaapiDecoderJPEG decoder;
     VideoConfigBuffer config;
     VideoDecodeBuffer buffer;
 
     config.flag = 0;
 
-    buffer.data = const_cast<uint8_t*>(g_SimpleJPEG.data());
-    buffer.size = g_SimpleJPEG.size();
+    buffer.data = const_cast<uint8_t*>(data.data());
+    buffer.size = data.size();
     buffer.timeStamp = 0;
 
     ASSERT_EQ(YAMI_SUCCESS, decoder.start(&config));
     ASSERT_EQ(YAMI_DECODE_FORMAT_CHANGE, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
 
-    EXPECT_TRUE(decoder.getOutput());
+    SharedPtr<VideoFrame> output = decoder.getOutput();
+    ASSERT_TRUE(output);
+    EXPECT_EQ(fourcc, output->fourcc);
+}
+
+VAAPIDECODER_JPEG_TEST(Decode_Simple)
+{
+    decodeJpeg(g_SimpleJPEG, YAMI_FOURCC_444P);
+    decodeJpeg(g_JPEG420, YAMI_FOURCC_IMC3);
+    /* this will trigger bug in dirver,
+     * see https://bugs.freedesktop.org/show_bug.cgi?id=97633c for details
+     */
+    /*
+    decodeJpeg(g_JPEG422H, YAMI_FOURCC_422H);
+    decodeJpeg(g_JPEG422V, YAMI_FOURCC_422V);
+    decodeJpeg(g_JPEG444P, YAMI_FOURCC_444P);
+    */
```

Comment:
Please use a [Value-Parameterized Test](https://github.com/google/googletest/blob/master/googletest/docs/V1_7_AdvancedGuide.md#how-to-write-value-parameterized-tests) here instead.

You will probably need to create a different test class for this.  See [vaapiencoder_jpeg_unittest.cpp](https://github.com/01org/libyami/blob/apache/encoder/vaapiencoder_jpeg_unittest.cpp) for an example of a parameterized test.